File: src/main/java/com/mm/moneymanager/model/debt/Debt.java
package com.mm.moneymanager.model.debt;

import com.mm.moneymanager.model.Audit.UserDateAudit;
import com.mm.moneymanager.model.user.User;
import lombok.*;

import javax.persistence.*;
import java.math.BigInteger;
import java.time.LocalDate;

@EqualsAndHashCode(callSuper = false)
@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Debt extends UserDateAudit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String company;
    BigInteger amount;

    LocalDate dueDate;

    @ToString.Exclude
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: src/frontend/src/app/model/debt.ts
export class Debt {

  company: string;
  amount: number;
  dueDate: string;
  id: number;


  constructor(company: string, amount: number, dueDate: string) {
    this.company = company;
    this.amount = amount;
    this.dueDate = dueDate;
  }


}
